### PR TITLE
dns: T4509: Add dns64-prefix option

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

rfc6147: DNS Extensions for Network Address Translation
         from IPv6 Clients to IPv4 Servers

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4509

## Component(s) name

dns
## How to test
```
set service dns forwarding dns64-prefix 2001:db8:aabb::/96
```
Expected conf:
```
vyos@r14# cat /run/powerdns/recursor.conf | grep pref
# dns64-prefix
dns64-prefix=2001:db8:aabb::/96
[edit]
vyos@r14# 

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/dns-forwarding.xml.in [hunk 1]
```diff
@@ -36,6 +36,18 @@
...
               </leafNode>
+              <leafNode name="dns64-prefix">
+                <properties>
+                  <help>Help to communicate between IPv6-only client and IPv4-only server</help>
+                  <valueHelp>
+                    <format>ipv6net</format>
+                    <description>IPv6 address and prefix length</description>
+                  </valueHelp>
+                  <constraint>
+                    <validator name="ip-prefix"/>
```
Comment (c-po) [inline interface-definitions/dns-forwarding.xml.in]: I guess it should be `ipv6-prefix`?

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/dns-forwarding.xml.in (hunk 1 above)
Comment (sever-sever) [reply interface-definitions/dns-forwarding.xml.in]: > I guess it should be `ipv6-prefix`?

Exactly!
Fixed

Comment (c-po) [reply interface-definitions/dns-forwarding.xml.in]: PDNS man page states: `Enable DNS64 (RFC 6147) support using the supplied /96 IPv6 prefix`. We probably should use this help string. Also what happens if a /64 is supplied? Does PDNS terminate? If this is the case, please add a `verify()` step to check prefix length.

Comment (sever-sever) [reply interface-definitions/dns-forwarding.xml.in]: With /64 it treminated `code=exited, status=1/FAILURE`
I'll add checks

Comment (c-po) [reply interface-definitions/dns-forwarding.xml.in]: Please also add Smoketests verifying these checks.

To test if an exception is raised use:
```python
with self.assertRaises(ConfigSessionError):
    self.cli_commit()
```

Comment (sever-sever) [reply interface-definitions/dns-forwarding.xml.in]: > Please also add Smoketests verifying these checks.

Done!

**Pull request merged**